components/search/BookingInfoSearch/BookingInfoSearchUIComplete.tsx

```diff
@@ -1,57 +1,97 @@
-};
+import { searchParamsType } from "@/app/page";
 
 type BookingInfoUISearchCompleteProps = {
-  bookingInfo: BookingInfo;
-  setBookingInfo: (info: BookingInfo) => void;
+  bookingInfo: searchParamsType;
+  setBookingInfo: (info: searchParamsType) => void;
 };
 
-const BookingInfoUISearchComplete: React.FC<BookingInfoUISearchCompleteProps> = ({
-  bookingInfo,
-  setBookingInfo,
-}) => {
+const BookingInfoUISearchComplete: React.FC<
+  BookingInfoUISearchCompleteProps
+> = ({ bookingInfo, setBookingInfo }) => {
+  const [tempBookingInfo, setTempBookingInfo] = useState(bookingInfo);
+  const [isDateValid, setIsDateValid] = useState(true);
+  const [isValid, setIsValid] = useState(true);
+
   const handleDateChange = (dates: {
     checkinDate: string;
     checkoutDate: string;
     numDays: number;
   }) => {
-    setBookingInfo({ ...bookingInfo, ...dates });
+    setTempBookingInfo((prev) => ({ ...prev, ...dates }));
+  };
+
+  const handleAdultsNumberChange = (adults: number | null) => {
+    if (adults !== null) {
+      setIsValid(true);
+      setTempBookingInfo((prev) => ({ ...prev, adultsNumber: adults }));
+    } else {
+      setIsValid(false);
+    }
   };
 
-  const handleAdultsNumberChange = (adults: number) => {
-    setBookingInfo({ ...bookingInfo, adultsNumber: adults });
+  const handleApplyFilters = () => {
+    if (isValid) {
+      setBookingInfo(tempBookingInfo);
+    }
+  };
+
+  const handleRevertChanges = () => {
+    const confirmRevert = window.confirm(
+      "Are you sure you want to revert your booking changes?"
+    );
+    if (confirmRevert) {
+      setTempBookingInfo(bookingInfo);
+      setBookingInfo(tempBookingInfo);
+    }
   };
 
   return (
-    <div >
+    <div>
       <h2 className="text-2xl font-bold">Booking Information</h2>
-
       <div className="gap-5 join">
-        {/* Booking Date Range Input */}
-          <div className="join-item">
-            <h3 className="text-primary font-semibold">Booking Dates</h3>
-            <DatePickerWithRange
-              onChange={handleDateChange} // Update parent state when date range changes
-              defaultNumDays={bookingInfo.numDays} // Default number of days
-            />
-          </div>
-          {/* Adults Number Input */}
-          <div className="join-item">
-            <AdultsNumberInput
-              selectedNumber={bookingInfo.adultsNumber}
-              onChange={handleAdultsNumberChange}
-            />
-          </div>
+        <div className="join-item">
+          <h3 className="text-primary font-semibold">Booking Dates</h3>
+          <DatePickerWithRange
+            bookingInfo={bookingInfo}
+            onChange={handleDateChange}
+            onValidationChange={setIsDateValid}
+          />
+        </div>
+        <div className="join-item">
+          <AdultsNumberInput
+            selectedNumber={tempBookingInfo.adultsNumber}
+            onChange={handleAdultsNumberChange}
+          />
+          {!isValid && (
```

Comment:
Don't your template inputs handle this? If it's a hard coded error message you can pass in the error message as a prop.